ttnn/cpp/ttnn/operations/eltwise/ternary_backward/ternary_backward.hpp

```diff
@@ -26,32 +26,57 @@ struct ExecuteTernaryBackward {
 
      //Type 0: 3 inputs, 1 grad tensor, 1 float
 
-    static std::vector<ttnn::Tensor> execute_on_worker_thread(
+    static std::vector<Tensor> execute_on_worker_thread(
         const Tensor &grad_tensor_arg,
         const Tensor &input_tensor_a_arg,
         const Tensor &input_tensor_b_arg,
         const Tensor &input_tensor_c_arg,
         const MemoryConfig &memory_config) {
-        auto op_type = TernaryBackwardFunction::get_function_type0(ternary_backward_op_type);
+        auto op_type = get_function_type0<ternary_backward_op_type>();
         return op_type(grad_tensor_arg, input_tensor_a_arg, input_tensor_b_arg, input_tensor_c_arg, memory_config);
         }
+};
+
+template <TernaryBackwardOpType ternary_backward_op_type>
+struct ExecuteTernaryBackwardFloat {
+
+    static inline std::vector<Tensor> create_async_output_tensors(
+        const std::vector<Tensor> &input_tensors, const std::vector<std::optional<const Tensor>>& optional_inputs) {
+        const auto& input_tensor = input_tensors.at(0);
+        return {Tensor(operation::get_workers_for_op_output({input_tensor})),
+                                            Tensor(operation::get_workers_for_op_output({input_tensor})),
+                                            Tensor(operation::get_workers_for_op_output({input_tensor}))};
+    }
 
     //Type 1: 3 inputs, 1 grad tensor, 1 float
 
-    static std::vector<ttnn::Tensor> execute_on_worker_thread(
+    static std::vector<Tensor> execute_on_worker_thread(
         const Tensor &grad_tensor_arg,
         const Tensor &input_tensor_a_arg,
         const Tensor &input_tensor_b_arg,
         const Tensor &input_tensor_c_arg,
         float alpha,
         const MemoryConfig &memory_config) {
-        auto op_type = TernaryBackwardFunction::get_function_type(ternary_backward_op_type);
+        auto op_type = get_function_type1<ternary_backward_op_type>();
         return op_type(grad_tensor_arg, input_tensor_a_arg, input_tensor_b_arg, input_tensor_c_arg, alpha, memory_config);
         }
 
+};
+
+template <TernaryBackwardOpType ternary_backward_op_type>
+struct ExecuteTernaryBackwardOptional {
+
+    static inline std::vector<Tensor> create_async_output_tensors(
+        const std::vector<Tensor> &input_tensors, const std::vector<std::optional<const Tensor>>& optional_inputs) {
+        const auto& input_tensor = input_tensors.at(0);
```

Comment:
I don't like using .at call. If array is empty it will throw weird error. I'd check it separately. It would be nice to fix it here too.

Reply:
Might have to ask Akhmed about this. I am going under the assumption that an op cannot be executed without passing atleast one input tensor which is/are being passed here